prem_utils/connectors/base.py

```diff
@@ -19,6 +53,9 @@ def __init__(self, prompt_template: str = None):
             409: errors.PremProviderConflictError,
         }
 
+    def list_models(self):
```

Comment:
I don't like this. We are talking about connectors. We should not fetch that from a JSON. If we need the list models method, I would prefer that we send a request to OpenAI to get it.

Reply:
Yes then we can do that, I actually assumed that some service provider would have the method or not. But will investigate this and put into a different PR :)

Reply:
For example, I was checking out Cloudflare's documentation and I did not got any method that does it. [here](https://developers.cloudflare.com/workers-ai/models/text-generation/)

Reply:
Yep, exactly. Most of them don't provide it :) that's why we should not put it

Reply:
I see, okay I removed it